### PR TITLE
chore: toolbox conversion currencies + supported currencies update

### PR DESCRIPTION
**Pull request by yomarion**
## Description of the changes
* Upgraded to ethers v5
* Reshaped toolbox with yargs
* Added a util to get a currency hash
* Added generation of supportedCurrencies map

### CODE REVIEW

File: packages/toolbox/package.json [hunk 1]
```diff
@@ -36,16 +36,18 @@
...
+    "@requestnetwork/currency": "^0.5.0",
     "@requestnetwork/epk-signature": "0.5.31",
     "@requestnetwork/request-client.js": "0.36.0",
     "@requestnetwork/smart-contracts": "0.25.0",
     "@requestnetwork/types": "0.32.0",
     "axios": "0.21.1",
-    "ethers": "4.0.48",
-    "yargs": "16.2.0"
+    "ethers": "^5.1.0",
+    "yargs": "^16.2.0"
```
Comment (benjlevesque) [inline packages/toolbox/package.json:50]: fixed versions + consistency with other libs
```suggestion
    "ethers": "5.0.32",
    "yargs": "16.2.0"
```


Please remember to `yarn install` to fix the `yarn.lock` :)

File: packages/toolbox/src/chainlinkConversionPathTools.ts [hunk 2]
```diff
@@ -1,11 +1,36 @@
 import { ethers } from 'ethers';
 import { chainlinkConversionPath } from '@requestnetwork/smart-contracts';
+import yargs = require('yargs');
```
Comment (benjlevesque) [inline packages/toolbox/src/chainlinkConversionPathTools.ts:3]: if you add `yarn add --dev @types/yargs` you can import `yargs` properly

File: packages/toolbox/src/index.ts [hunk 3]
```diff
@@ -1,6 +1,19 @@
+import yargs = require('yargs');
```
Comment (benjlevesque) [inline packages/toolbox/src/index.ts:1]: same as above

File: packages/toolbox/src/index.ts [hunk 4]
```diff
@@ -1,6 +1,19 @@
...
   CreateRequest,
 };
+
+// With yargs, options can be passed as command line options (--my-option),
+//  or environment variables (RN_MY_OPTION)
+yargs
+  .env('RN')
+  .command(chainlinkAggregatorsCommandModule)
+  .command(currencyHashCommandModule)
+  .demandCommand().argv;
```
Comment (benjlevesque) [inline packages/toolbox/src/index.ts:19]: move to cli.ts

**Review by benjlevesque — CHANGES_REQUESTED** [review]

**Review by benjlevesque — APPROVED** [review]

File: packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts [hunk 5]
```diff
@@ -89,15 +101,16 @@ function createCreationAction(
 
   const network = creationParameters.network || 'mainnet';
   if (!supportedCurrencies[network]) {
-    throw Error('network not supported');
+    throw Error(`network ${network} not supported`);
```
Comment (benjlevesque) [inline packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts:104]: added details for debugging

File: packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts [hunk 6]
```diff
@@ -89,15 +101,16 @@ function createCreationAction(
...
   }
   const supportedErc20: string[] = supportedCurrencies[network][RequestLogicTypes.CURRENCY.ERC20];
-  if (
-    creationParameters.acceptedTokens.some(
-      (address) => !supportedErc20.includes(address.toLowerCase()),
-    )
-  ) {
-    throw Error('acceptedTokens must contain only supported token addresses (ERC20 only)');
+
```
Comment (benjlevesque) [inline packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts:107]: same as above

File: packages/toolbox/src/chainlinkConversionPathTools.ts [hunk 7]
```diff
@@ -38,49 +60,45 @@ class ChainlinkConversionPathTools {
...
+      this.contractChainlinkConversionPath.filters.AggregatorUpdated(),
+      this.chainlinkConversionPathCreationBlockNumber,
+      'latest',
+    );
 
     // Parses, filters and creates the events from the logs with the payment reference
     const aggregatorsMaps = logs.reduce(
       // Map: Input currency => Output currency => aggregator address
-      (aggregators: Map<string, Map<string, string>>, log: any) => {
+      (aggregators, log) => {
```
Comment (benjlevesque) [inline packages/toolbox/src/chainlinkConversionPathTools.ts:74]: ethers v5 changes

File: yarn.lock [hunk 8]
```diff
@@ -8971,6 +8963,21 @@ ethers@5.0.32:
     "@ethersproject/web" "5.0.14"
     "@ethersproject/wordlists" "5.0.10"
 
+ethers@^4.0.0-beta.1, ethers@^4.0.32:
```
Comment (vrolland) [inline yarn.lock:8966]: Why do we have ethers 4 again ?

**Review by vrolland — APPROVED** [review]

File: yarn.lock (hunk 8 above)
Comment (benjlevesque) [reply yarn.lock:8966]: web3 and truffle depend on it apparently
This will be fixed by another PR where I remove web3 😁

**Pull request merged**